src/LinearSolve.jl

```diff
@@ -143,6 +147,46 @@ end
     include("factorization_sparse.jl")
 end
 
+# Solver Specific Traits
+## Needs Square Matrix
+"""
+    needs_square_A(alg)
+
+Returns `true` if the algorithm requires a square matrix.
+
+Note that this checks if the implementation of the algorithm needs a square matrix by
+trying to solve an underdetermined system. It is recommended to add a dispatch to this
+function for custom algorithms!
+"""
+needs_square_A(::Nothing) = false  # Linear Solve automatically will use a correct alg!
+function needs_square_A(alg::SciMLLinearSolveAlgorithm)
```

Comment:
Just remove the fallback please.

Reply:
Or rather, the fallback is true. That's a safe fallback, methods can opt-in to supporting non-square, but anything that supports non-square supports square so that's a safe assumption.